### PR TITLE
ARROW-5686: [R] Review R Windows CI build

### PR DESCRIPTION
**Pull request by nealrichardson**
I reviewed with @jeroen and he said "That seems about right." This patch reorganizes the script and adds/updates comments to reflect a better understanding of the process.

### CODE REVIEW

**Comment by nealrichardson** [comment]
There was a transient failure on Appveyor here but it passed on my fork just now: https://ci.appveyor.com/project/nealrichardson/arrow/builds/26893899

File: ci/windows-pkg-arrow-for-r.sh [hunk 1]
```diff
@@ -28,39 +28,51 @@ mkdir $DST_DIR
 # Grab the headers from one, either one is fine
 mv mingw64/include $DST_DIR
 
-# Move the 64-bit versions into the expected location
+# Make the rest of the directory structure
+# lib-4.9.3 is for libraries compiled with gcc 4.9 (Rtools 3.5)
 mkdir -p $DST_DIR/lib-4.9.3/x64
+mkdir -p $DST_DIR/lib-4.9.3/i386
+# lib is for the new gcc 8 toolchain (Rtools 4.0)
```
Comment (pitrou) [inline ci/windows-pkg-arrow-for-r.sh]: gcc 8, really? Does that mean we can ditch gcc 4.9 at some point?

Comment (jeroen) [reply ci/windows-pkg-arrow-for-r.sh]: No :) We want to keep supporting that too.

File: ci/windows-pkg-arrow-for-r.sh [hunk 2]
```diff
@@ -28,39 +28,51 @@ mkdir $DST_DIR
...
+mkdir -p $DST_DIR/lib/x64
+mkdir -p $DST_DIR/lib/i386
```
Comment (fsaintjacques) [inline ci/windows-pkg-arrow-for-r.sh]: For future reference, `mkdir -p $DEST_DIR/lib{,-4.9.3}/{x64,i386}`

**Pull request closed without merging**